desktop/tests/test_provider_conformance.py

```
@@ -0,0 +1,1460 @@
+def test_stream_chunk_rejects_ambiguous_mixed_payloads():
+    usage = provider_module.ModelUsage(1, 1, 2, 0, 1, 0)
+
+    with pytest.raises(ValueError, match="exactly one lifecycle payload"):
+        provider_module.StreamChunk(text_delta="answer", usage=usage)
+
+
+def test_model_usage_rejects_negative_or_inconsistent_counts():
+    with pytest.raises(ValueError, match="non-negative"):
+        provider_module.ModelUsage(-1, 1, 0, 0, 0, 0)
+
+    with pytest.raises(ValueError, match="cached.*uncached"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)